driver-core/src/main/java/com/datastax/driver/core/Cluster.java

```diff
@@ -2043,8 +2084,11 @@ public void run() {
                         schemaInAgreement = ControlConnection.waitForSchemaAgreement(connection, Manager.this);
                         if (!schemaInAgreement)
                             logger.warn("No schema agreement from live replicas after {} s. The schema may not be up to date on some nodes.", configuration.getProtocolOptions().getMaxSchemaAgreementWaitSeconds());
-                        if (refreshSchema)
-                            ControlConnection.refreshSchema(connection, keyspace, table, Manager.this, false);
+                        if (refreshSchema) {
+                            submitSchemaRefresh(keyspace, table);
+                            // force immediate event delivery
+                            schemaRefreshRequestDebouncer.deliverEvents();
```

Comment:
This tries to mimic the old behavior where the refresh was done synchronously in the calling thread.